native/src/accelerated_bn/utils.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Generic executions of the operations for *Arkworks* elliptic curves.
+
+// As not all functions are used by each elliptic curve and some elliptic
+// curve may be excluded by the build we resort to `#[allow(unused)]` to
+// suppress the expected warning.
+
+use super::BnCryptoError;
+use ark_ec::{
+    pairing::{MillerLoopOutput, Pairing},
+    short_weierstrass::{Affine as SWAffine, Projective as SWProjective, SWCurveConfig},
+    CurveConfig, VariableBaseMSM,
+};
+use ark_scale::{
+    ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Compress, Validate},
+    scale::{Decode, Encode},
+};
+
+// SCALE encoding parameters shared by all the enabled modules
+const SCALE_USAGE: u8 = ark_scale::make_usage(Compress::No, Validate::No);
```

Comment:
What's the `Validate` doing ? We are assuming EC points have already been validated previously ?

Reply:
Hmm, from the documentation it would seem that Validate only comes into play when you deserialize. Specifically,
[](https://github.com/w3f/ark-scale/blob/master/src/lib.rs#L115)
I assume as much w.r.t. EC points. Polkadot-SDK seems to be making this assumption as well.